habitat-hitl/habitat_hitl/environment/controllers/llm_controller.py

```diff
@@ -0,0 +1,167 @@
+import logging
+import threading
+from typing import Any, Dict, Union
+
+import numpy as np
+from habitat_llm.agent import Agent
+from habitat_llm.agent.env import EnvironmentInterface
+from habitat_llm.planner.llm_planner import LLMPlanner
+from habitat_llm.utils import fix_config, setup_config
+from habitat_llm.utils.analysis import CodeTimer
+from hydra.utils import instantiate
+from omegaconf import DictConfig
+
+import habitat
+import habitat.config
+from habitat.core.environments import GymHabitatEnv
+from habitat_hitl.environment.controllers.baselines_controller import (
+    SingleAgentBaselinesController,
+)
+
+
+class LLMController(SingleAgentBaselinesController):
+    """Controller for single LLM controlled agent."""
+
+    def __init__(
+        self,
+        agent_idx: int,
+        is_multi_agent: bool,
+        config: DictConfig,
+        gym_habitat_env: GymHabitatEnv,
+    ):
+        self._config = config
+        self._is_multi_agent = is_multi_agent
+        self._gym_habitat_env = gym_habitat_env
+        self._habitat_env = gym_habitat_env.unwrapped.habitat_env
+        self._agent_idx = agent_idx
+        # TODO: gather this from config
+        self._agent_action_length = 28
+        self._thread: Union[None, threading.Thread] = None
+        self._low_level_actions: Union[None, dict, np.ndarray] = {}
+        self._iter = 0
+        self._skip_iters = 0
+        logging.basicConfig(
+            filename="/home/priyamp/hitl/act_timing.log",
+            filemode="a",
+            format="%(asctime)s,%(msecs)d %(name)s %(levelname)s %(message)s",
+            datefmt="%H:%M:%S",
+            level=logging.DEBUG,
+            force=True,
+        )
+        self._analysis_logger = logging.getLogger("LLMController")
+        self._analysis_logger.debug("LLMController initialized")
+
+        with habitat.config.read_write(self._config):
+            fix_config(self._config)
+        seed = 47668090
```

Comment:
Where does this seed come from?

Reply:
It's the default in all our code. We can move this to config. But same as what other experiments are using.